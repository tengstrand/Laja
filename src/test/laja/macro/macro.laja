
#macro mymacro(a)
Input parameter a = {a}
Variable b = {b}##
#end

#macro empty()
abc##
#end

#macro for(expression)
  #foreach (i in [1,2,3])
{expression}
  #end
#end

#write "target/test-output/macro.out"
  #set (b = "bbb")
{mymacro(123)}
{for("i = {i}")}##
{empty()} {empty()}
#end
