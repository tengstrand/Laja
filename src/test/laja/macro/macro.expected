Input parameter a = 123
Variable b = bbb
i = 1
i = 2
i = 3
abc abc
